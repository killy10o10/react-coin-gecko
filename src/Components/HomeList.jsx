import { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { showCountries } from "../Redux/home/home";
import Home from "./Home";
import Search from "./Search";

const HomeList = () => {
  const [query, setQuery] = useState('');

  const onQuerySearch = (event) => {
    setQuery(event.target.value);
  };
  const countriesArray = useSelector((state) => state.country);
  const filterCountries = countriesArray.data.filter((country) => (
    country.name.toLowerCase().includes(query.toLowerCase())
  ));
  const dispatch = useDispatch();
  useEffect(() => {
    if (countriesArray.data.length === 0) {
      dispatch(showCountries());
    }
  }, []);

  return (
    <section className="country-section-container">
      <Search
        query={query}
        onQuerySearch={onQuerySearch}
      />
      <section className="country-section">
        {filterCountries.map((data) => (
          <Home
            key={data.name}
            country={data}
          />
        ))}
      </section>
    </section>

  );
};

export default HomeList;
