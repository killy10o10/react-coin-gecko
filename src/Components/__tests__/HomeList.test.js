import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import HomeList from '../HomeList';
import store from '../../Redux/configureStore';

it('Test HomeList renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <HomeList />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
