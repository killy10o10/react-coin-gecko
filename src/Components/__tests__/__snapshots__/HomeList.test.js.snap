// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HomeList renders correctly 1`] = `
<section
  className="country-section-container"
>
  <label
    className="search-label"
    htmlFor="search"
  >
    <input
      className="search-bar"
      id="search"
      onChange={[Function]}
      placeholder="Search Country"
      type="text"
      value=""
    />
    <i
      className="bi bi-search"
    />
  </label>
  <section
    className="country-section"
  />
</section>
`;
